presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -377,9 +377,6 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
             if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.APPEND) {
                 // Append to an existing partition
                 checkState(!immutablePartitions);
-                if (bucketNumber.isPresent()) {
```

Comment:
To be honest I was expecting something unexpected here. Glad it works.

Reply:
ha! so it was a trap?
like ["Bad error message"](https://github.com/prestodb/presto/issues/7030) issue I was doing for 3 months?

Reply:
But you learned that much...